src/hooks/stores/useSyncStore.ts

```diff
@@ -0,0 +1,102 @@
+import { useMemo } from "react";
+import { create } from "zustand";
+
+type Progress = {
+  completed: number;
+  total: number;
+};
+
+type ActiveSync = {
+  progress: Progress;
+  syncIsFinished: boolean;
```

Comment:
it is derived, but I added it as a easy way to determine if `completed === total`. Since state is manually set in the hook, we can guarantee it is derived. In other words, due to the implementation, there is no way a developer can change the state of `syncIsFinished` unless they change the implementation.